### DISCUSSION TITLE
Any webrtc implementation in avalonia?

### DISCUSSION THREAD

**Discussion by aloksharma1** [AvaloniaUI/Avalonia#16502] [General]
i would like to know if someone has implemented this before in avalonia or if its feasible in c# ecosystem with avalonia or similar tools. my requirement is to make audio/video rtc using webrtc, sockets or similar technology, i have used signalR for text chats and its seamless but cant find anything that can help with audio/video.

thanks for the help

**Comment by LaurentInSeattle** [comment]
I know about this one: https://ironpdf.com/blog/net-help/csharp-webrtc/ 
Not cheap.
So, yes it can be done...

**Reply by aloksharma1** [reply]
thanks although its not what i want, it shows the possibility how it can be done, i will check it out.

**Reply by KieranFoot** [reply]
Has nothing to do with WebRTC... It shows using IronPdf with WebRTC, but IronPDF has nothing to do with the WebRTC side of things, so as always, WebRTC itself remains free.

**Comment by thevortexcloud** [comment]
Avalonia supports Web assembly. As such if you are going to target web assembly you can in theory just reuse all the existing browser APIs and embed the webcam and what not as a native control in Avalonia. There is some sample code for embedding a browser element here:

https://github.com/AvaloniaUI/Avalonia/blob/78935ba85c4400e781edd821e47be30a9f5d9b6f/samples/ControlCatalog.Browser/EmbedSample.Browser.cs

**Reply by aloksharma1** [reply]
ok and what should i use for desktop application?

**Reply by thevortexcloud** [reply]
That's trickier. Some quick searches however shows this is an option:

https://github.com/sipsorcery-org/sipsorcery

No idea how well it works though. Otherwise you should in theory be able to use some native libraries and interop with them from C#. EG

https://webrtc.github.io/webrtc-org/native-code/native-apis/
